File: src/main/Kotlin/boj_step21/boj11729.java
import java.io.*;

public class boj11729 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        int k = (int) Math.pow(2, n) - 1;
        StringBuilder tmp = new StringBuilder(String.valueOf(k)).append("\n");
        hanoi(n, 1, 3, tmp);
        writer.append(tmp).close();
    }

    private static void hanoi(int n, int from, int to, StringBuilder tmp) throws IOException {
        if (n == 0) return;
        int spare = 6 - from - to;
        hanoi(n - 1, from, spare, tmp);
        tmp.append(from).append(" ").append(to).append("\n");
        hanoi(n - 1, spare, to, tmp);
    }
}

File: src/main/Kotlin/boj_step21/boj11729.kt

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    val k = Math.pow(2.0, n.toDouble()).toInt() - 1
    val b = StringBuilder().appendLine(k)
    hanoi(n, 1, 3, b)
    w.append(b).close()
}

private fun hanoi(n: Int, from: Int, to: Int, b: StringBuilder) {
    if (n == 0) return
    val spare = 6 - from - to
    hanoi(n - 1, from, spare, b)
    b.append(from).append(" ").append(to).appendLine()
    hanoi(n - 1, spare, to, b)
}